tournament_results.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+"""
+Get the results a Challonge tournament and use the information to update
+the Smash results spreadsheet.
+
+Pre-Reqs:
+1. Set up OAuth: https://developers.google.com/drive/api/v3/quickstart/python
+2. Download the "My Project*.json" authentication file (this is the value for
+   the --oauth parameter)
+3. Share the results spreadsheet with with "Edit" rights to the "client_email"
+   listed in the OAuth file. Should be "@*.iam.gserviceaccount.com".
+4. Reflect briefly on how much more work this was than typing in the results
+   by hand.
+5. Run the script
+
+"""
+import argparse
+import challonge
+import configparser
+import gspread
+from oauth2client.service_account import ServiceAccountCredentials
+import os
+import re
+from requests.exceptions import HTTPError
+import sys
+
+import defaults
+import util_challonge
+
+if __name__ == "__main__":
+    argparser = argparse.ArgumentParser(
+                    description="Parse the results of a Challonge tournament, "
+                                "and post the results to a Google Sheet. "
+                                "Make sure to do the work in the header of "
+                                "this file before running it.",
+                    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+                )
+    argparser.add_argument(
+        "tourney_num_file",
+        default='TOURNEY_NUM',
+        help="file containing what tournament we're on",
+    )
+    argparser.add_argument(
+        "--config_file",
+        default=defaults.DEFAULT_CONFIG_FILENAME,
+        help="the config file to read your Challonge credentials from",
+    )
+    argparser.add_argument(
+        "--oauth",
+        default="client_secret.json",
+        help="OAuth JSON file to access the Drive API"
+    )
+
+    args = argparser.parse_args()
+
+    # We need to initialize our Challonge credentials before we can
+    # make any API calls.
+    initialized = util_challonge.\
+        set_challonge_credentials_from_config(args.config_file)
+    if not initialized:
+        sys.exit(1)
+
+    if not os.path.isfile(args.oauth):
+        print("JSON authentication file '{}' not found to access Sheets.\n"
+              "Make sure you followed the OAuth steps in the header to\n"
+              "download the client secret file.".format(args.oauth))
+        sys.exit(1)
+
+    config_parser = configparser.RawConfigParser()
+    config_parser.read(args.config_file)
+
+    try:
+        SPREADSHEET_KEY = config_parser.get('Sheets', 'spreadsheet_key')
+    except configparse.Error as err:
+        sys.stderr.write("Be sure to set 'spreadsheet_key' in config file: {}"
+                         .format(err))
+        sys.exit(1)
+
+    # Read previous tournament number
+    with open(args.tourney_num_file) as f:
+        TOURNEY_NUM = int(f.read()) + 1
+
+    tourney_name = "mtvmelee{}".format(TOURNEY_NUM)
+
+    try:
```

Comment:
You can do:

```
tourney_info = util_challonge.get_tourney_info(tourney_name)
if not tourney_info:
    # ...
```

This ensures that the HTTPError code is 404 before returning None (raising any other error so the user knows if, say, they're disconnected from the internet) and abstracts away the networking layer underlying the challonge Python API (which has changed between different versions of the API)